turbo/app/snapshots_cmd.go

```diff
@@ -1396,11 +1396,16 @@ func doRetireCommand(cliCtx *cli.Context, dirs datadir.Dirs) error {
 		return err
 	}
 	blockReader, _ := br.IO()
-	from2, to2, ok := freezeblocks.CanRetire(to, blockReader.FrozenBlocks(), coresnaptype.Enums.Headers, nil)
+	frozenBlocks := blockReader.FrozenBlocks()
+	if br.ChainConfig.Bor != nil {
+		frozenBlocks = min(frozenBlocks, blockReader.FrozenBorBlocks())
+	}
+	maxAvailableBlock := to
+	from2, to2, ok := freezeblocks.CanRetire(to, frozenBlocks, maxAvailableBlock, coresnaptype.Enums.Headers, nil)
```

Comment:
and i see first parameter here is stage_senders progress. and this parameter called `curBlockNum` - which looks similar to `maxAvailableBlock`.